### PR TITLE
Add Transaction Support

### PR DESCRIPTION
**Pull request by mvriel**

### CODE REVIEW

File: src/Transaction.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace PhpInPractice\Matters\Aggregate;
+
+interface Transaction
+{
+    public function push($streamUri, array $writeableEvents);
+    public function flush();
```
Comment (fvdb) [inline src/Transaction.php]: Usually the term "commit" is used to to commit a transaction

**Comment by ussuritiger** [comment]
:+1:

**Comment by rvounik** [comment]
:+1:

File: src/Transaction/TransactionException.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace PhpInPractice\Matters\Aggregate\Transaction;
+
+class TransactionException extends \RuntimeException
+{
+
+}
```
Comment (fvdb) [inline src/Transaction/TransactionException.php:8]: This base class seems a bit premature

File: src/Transaction/SingleStream.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        $this->streamUri = $streamUri;
+        $this->events = array_merge($this->events, $writeableEvents);
+    }
+
+    public function commit()
+    {
+        $collection = new WritableEventCollection($this->events);
+        $this->eventstore->writeToStream($this->streamUri, $collection);
```
Comment (fvdb) [inline src/Transaction/SingleStream.php:40]: You are not emptying $this->events after the write

File: src/Transaction/TransactionException.php (hunk 2 above)
Comment (mvriel) [inline src/Transaction/TransactionException.php:8]: It is a little bit; but I wanted to provide third-parties with a common base to work with

File: src/Transaction/SingleStream.php (hunk 3 above)
Comment (fvdb) [inline src/Transaction/SingleStream.php:40]: You are also not resetting $this->streamUri

**Comment by fvdb** [comment]
`final` missing at several places

File: src/Transaction/TransactionException.php (hunk 2 above)
Comment (fvdb) [inline src/Transaction/TransactionException.php:8]: Seems rather hypothetical, considering that there is only a single exception

Comment (fvdb) [inline src/Transaction/TransactionException.php:8]: I don't really like code that doesn't serve a concrete use case, but I'm 2/10 on this, so if you really want to keep this in I'm fine with that

**Comment by mbed67** [comment]
:+1:

**Pull request merged**